### PR TITLE
Add tests for decoding

### PR DESCRIPTION
**Pull request by KLBa**
Add some unit tests inspired by #21

Katrin Bauer <[katrin.b.bauer@daimler.com](mailto:katrin.b.bauer@daimler.com)>, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/daimler-foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: examples/somersaultlazy.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ async def main(args):
             tester_task = asyncio.create_task(tester_main(network))
 
         if args.mode == "server":
-            await tester_task
```
Comment (andlaus) [inline examples/somersaultlazy.py:339]: good catch. Thanks!

File: odxtools/decodestate.py [hunk 2]
```diff
@@ -11,9 +11,9 @@ class ParameterValuePair(NamedTuple):
 
 class DecodeState(NamedTuple):
     """Utility class to be used while decoding a message."""
-    # bytes to be decoded
     coded_message: Union[bytes, bytearray]
-    # values of already decoded parameters
+    """bytes to be decoded"""
```
Comment (andlaus) [inline odxtools/decodestate.py:15]: these are doctrings for IDEs/linters, right? (i.e. python completely ignores them?)

**Review by andlaus — APPROVED** [review]
nice! (as always)

File: odxtools/decodestate.py (hunk 2 above)
Comment (KLBa) [reply odxtools/decodestate.py:15]: Yup, there is no `__doc__` or `help(...)` for attribute doc strings but, e.g., VSCode supports them and shows them on hover.

**Pull request merged**